include/exadg/solvers_and_preconditioners/multigrid/multigrid_preconditioner_base.cpp

```diff
@@ -306,10 +306,105 @@ MultigridPreconditionerBase<dim, Number>::check_levels(std::vector<MGLevelInfo>
   }
 }
 
+/**
+ * Helper function for creating a geometric coarsening sequence: Replicate a
+ * parallel::distributed::Triangulation completely in a serial
+ * triangulation. This can potentially be very memory-consuming, so the
+ * triangulation is only filled on the first rank of a compute node, i.e.,
+ * those MPI ranks where `rank % n_processes_per_node` is zero.
+ */
+template<int dim, int spacedim>
+void
+replicate_distributed_triangulation_on_node(
```

Comment:
I like this refactoring!

I wonder if ordering the parameters according to the `dst`, `src` logic (widely used in dealii or in ExaDG) would be better. Further, the word `serial` in the function name would also help. `serialize_distributed_triangulation_per_node()`? Not sure if this is better?

Reply:
I decided against `serialize_...` because that has a different meaning in terms of storing a stream of bytes so that might potentially confuse the reader. I opted for name `gather_distributed_triangulation_by_node` as the name `gather` bears exactly the meaning of replicating all information in the MPI context.